### ISSUE TITLE
Power should not accept arrays with requires_grad=True as exponents

### ISSUE THREAD

**Issue by andykee** [andykee/loupe#1]
### Description
Since `power()` doesn't support computing the gradient of exponents, it should throw an error if the exponent is a loupe array with `requires_grad=True`.

### Expected behavior
Throw a `RuntimeError` if the exponent has `requires_grad=True`.

### Actual behavior
No error is thrown.
